### DISCUSSION TITLE
Expose notification mark as done functionality through API

### DISCUSSION THREAD

**Discussion by NoNameProvided** [community/community#15669] [Other Feature Feedback, Questions, & Ideas] [answered]
Currently, there is no way to mark a notification as done through the Github API. I would like to be able to mark a notification as "done" not just as "read". 

Context: We use Dependabot on all or non-archived repositories and the lack of grouped updates for Dependabot results in flowing my notification view with PRs from Dependabot. I wanted to create a Github OAuth app that checks for PR notifications from Dependabot and automatically marks them as done, but that functionality is not exposed in the current API.

**Comment by remal** [comment]
Same for me, but I use Renovate. I have really good test coverage and would like **not** to be bored by notifications of Renovate PRs.

**Comment by sgarbesi** [comment]
Gathering my notifications in a secondary application, this would be nice to have that application dismiss those notifications in GitHub.

**Comment by sparky-raccoon** [comment]
Reviving this question if relevant. Working on cronjobs to perform clean tasks, and would love to mark notifications as done (/notifications/beta/archive) + unsubscribe from further updates (/notifications/beta/unsubscribe).

**Comment by JohnCampionJr** [comment]
It would be very helpful if this was added....

**Reply by jonny-rimek** [reply]
cries in 1312 unread notifications

**Comment by rmacklin** [comment]
This API has been introduced: https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#mark-a-thread-as-done

However, there still doesn't seem to be an API parameter to filter notifications using `is:done` (or not) like we can in the web UI. This makes it difficult to actually make use of the "mark as done" API for integrations like gh-notify: https://github.com/meiji163/gh-notify/issues/47#issuecomment-1974962747

**Comment by Bambobee** [comment]
yeah

**Comment by mgroth0** [comment]
Yup. Just found this thread because I discovered I could not filter notifications from the API based on their "Done" status.

In fact, not only can we not filter by "Done" status in the request, but the returned data itself seems to also not contain any any information about "Done" status. So there is no workaround at all? There is absolutely no way to get all notifications from github that are not marked as "Done"?

**Comment by mkurz** [comment]
It would be great if the done flag would be added, but not just as boolean but as timestamp, so we could see, when an issue was marked as done (this would help to find notifications that have been marked as done by accident...)

**Comment by Alwaysin** [comment, marked as answer]
This is now possible as of API version 2026-03-10: https://docs.github.com/en/rest/activity/notifications?apiVersion=2026-03-10#mark-a-thread-as-done

**Reply by tyriis** [reply]
only 4 years :) thank you so much will save me and some people i know many hours a month

**Reply by rmacklin** [reply]
It's not new, see https://github.com/orgs/community/discussions/15669#discussioncomment-8658942 which also mentions a related gap
